### PR TITLE
HTTP Proxy setup for AWS

### PR DESCRIPTION
**Pull request by CaitlinOCallaghan**
## Features
- Setup AWS Cli and bucket/container policies
- Launch an HTTP proxy to upload streams to AWS S3 or MediaStore
- Purge MediaStore container in one go using a bash script

## Testing
- Listening HTTP proxy accepting and uploading files to MediaStore
image

- MediaStore results
image
image

- Container purge
image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,3 +25,107 @@ With the test content downloaded, you may now run the included demos. This will
...
+
+```console
+foo@bar:~$ aws configure
+```
+
+To ensure that your credentials have been added, checkout your aws directory and peek into the credentials file. 
+
+```console
+foo@bar:~$ cd ~/.aws
+foo@bar:~/.aws$ cat credentials
```
Comment (stephenlindauer) [inline README.md]: Minor suggestion: you can collapse this into a one liner

```suggestion
foo@bar:~$ cat ~/.aws/credentials
```

File: demo/server/emptyMediaStoreContainer.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+# WARNING: This script will completely empty your MediaStore container!
+
+containerName='shaka_ull'
```
Comment (stephenlindauer) [inline demo/server/emptyMediaStoreContainer.sh]: I believe the style guide for constants should be all uppercase with underscores:
ie.. `CONTAINER_NAME='shaka_ull'`

File: demo/server/launchAwsProxy.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+port=8080
+awsRegion='us-west-2'
+bucketName='shaka_ull'
+cloudStorage='mediastore' # Please select 'mediastore' or 's3'
+
+serverDir=$PWD
+s3UploadProxyDir='../../s3-upload-proxy'
```
Comment (stephenlindauer) [inline demo/server/launchAwsProxy.sh]: Same, uppercase these

File: demo/server/purge_mediastore_container.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+import boto3
+import sys
+from pprint import pprint
+
+filesToDelete = []
+foldersToRecurse = []
+
+def usage():
+
```
Comment (stephenlindauer) [inline demo/server/purge_mediastore_container.py:39]: remove empty line

File: demo/server/purge_mediastore_container.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+  print("%s [ContainerName] [Path]/" % sys.argv[0])
+  print("Example: %s TestContainerPleaseIgnore PathToDelete/" % sys.argv[0])
+
+def delete_things():
+  print("%s files to delete." % len(filesToDelete))
+  while len(filesToDelete) > 0:
+    thisFile = filesToDelete.pop()
+    print("=== DELETE:  %s ===" % thisFile)
+    try:
+      delete_response = object_client.delete_object(Path="%s" % (thisFile))
```
Comment (stephenlindauer) [inline demo/server/purge_mediastore_container.py]: do we need to store `delete_response` here?

File: demo/server/purge_mediastore_container.py [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+    except KeyboardInterrupt:
+      break
+      sys.exit
```
Comment (stephenlindauer) [inline demo/server/purge_mediastore_container.py]: Do we need the `break` here? sys.exit() should handle that

**Review by stephenlindauer — COMMENTED** [review]
Logic looks sound. Most comments are just on style/formatting.

One last suggestion would be to check out the `logging` module instead of using regular `print` calls for different verbosity. I haven't used it myself, but it seems like it could be useful for these types of scripts. 
https://docs.python.org/3.3/library/logging.html#module-logging

**Comment by CaitlinOCallaghan** [comment]
@stephenlindauer, I really like your idea for logging! Since the logging is a "nice to have" feature, I created [an issue ](https://github.com/CaitlinOCallaghan/ull-shaka-ecosystem/issues/6)for it and will implement it when I have some down time.

**Review by stephenlindauer — APPROVED** [review]

**Pull request merged**